### ISSUE TITLE
Unexpected "Couldn't determine your OS!" error

### ISSUE THREAD

**Issue by Roekeloosss** [matsn0w/IIS-EmergencyLS-ELS-FiveM#190]
[    c-scripting-core] Creating script environments for MISS-ELS
[     script:MISS-ELS] Home:    nil
[     script:MISS-ELS] HOMEPATH:        nil
[     script:MISS-ELS] SCRIPT ERROR: @MISS-ELS/server/init.lua:115: Couldn't determine your OS! Are your running on steroids??

**Comment by matsn0w** [comment]
That means the script couldn't determine whether you are running your server on Linux or Windows... What platform are you on?

**Comment by eriky1993** [comment]
got the same problem. running on Linux. 
Artifact 10188 works. something within the artifact changed.
Artifact 10188 is to old for other rescources

**Comment by matsn0w** [comment]
Closing in favor of #192

**Comment by Leavify** [comment]
I have this same issue, what to do to fix this?

**Comment by eriky1993** [comment]
Didn't fix it. Running now on artifact 11535 and hope for a update

**Comment by Leavify** [comment]
> Didn't fix it. Running now on artifact 11535 and hope for a update

Can you not go a couple artifacts higher? Or is that the highest possible in order for the ELS to work?

**Comment by eriky1993** [comment]
I didn't try it.

**Comment by Jakemammoet** [comment]
Can someone help me fix the [ script:MISS-ELS] SCRIPT ERROR: @MISS-ELS/server/init.lua:115: Couldn't determine your OS! Are your running on steroids?? issue i don't know how to fix it, i got a server with vibegames

**Comment by matsn0w** [comment]
I would appreciate it if someone could make a pull request for this, I don't have the time for that myself right now. Apparently something changed in FiveM that makes the OS check not work anymore.

**Comment by Lap76** [comment]
I was able to get it fixed with some help from chatGPT by changing the init.lua completely to the code i put in my fork. Currently working on the latest (unstable?) build 12882

https://github.com/matsn0w/MISS-ELS/pull/195

**Comment by Jakemammoet** [comment]
Does that fix the [ script:MISS-ELS] SCRIPT ERROR: @MISS-ELS/server/init.lua:115: Couldn't determine your OS! Are your running on steroids?? error?, if so can you help me fix it bc i don't really know anything about this script etc

**Comment by Lap76** [comment]
> Does that fix the [ script:MISS-ELS] SCRIPT ERROR: @MISS-ELS/server/init.lua:115: Couldn't determine your OS! Are your running on steroids?? error?, if so can you help me fix it bc i don't really know anything about this script etc

Yes it fixes the error with the current init.lua, you would just need to replace the old init.lua with my new one

**Comment by Jakemammoet** [comment]
I have put it in and it still gives me the error

**Comment by Lap76** [comment]
@Jakemammoet What file did you download, did you download the correct one from my pull request? Also what TX build/artifact  are you currently on?

I've tested it for fridays latest tx build/artifact 12882 and that one was working with my fix for the init.lua

**Comment by Jakemammoet** [comment]
I downloaded the IIS-EmergencyLS-ELS-FiveM and i'm on the same build as you

**Comment by Lap76** [comment]
Then idk why it isn't working for you, i've tested it on a fresh local server and for me its working on artifact 12882

a friend of mine is on artifact 12651 and it's working there just fine with the same file as in my pull request, i've created a release on the fix branch, maybe try downloading it through there? 

https://github.com/Lap76/IIS-EmergencyLS-ELS-FiveM/releases/tag/1.0.0

**Comment by Jakemammoet** [comment]
Hey, sorry i got really busy i'll try it today

**Comment by Trko0045** [comment]
[    c-scripting-core] script error in native 0000000076a9ee1f: vector::_M_default_append
[     script:MISS-ELS] SCRIPT ERROR: native 0000000076a9ee1f: vector::_M_default_append
[     script:MISS-ELS] > handler (@MISS-ELS/server/init.lua:123)
[           resources] Started resource MISS-ELS


i used the one you provided in the link. one at the top of the page and server artifacts 12746

**Comment by DevNettrix** [comment]
> [ c-scripting-core] script error in native 0000000076a9ee1f: vector::_M_default_append [ script:MISS-ELS] SCRIPT ERROR: native 0000000076a9ee1f: vector::_M_default_append [ script:MISS-ELS] > handler (@MISS-ELS/server/init.lua:123) [ resources] Started resource MISS-ELS
> 
> i used the one you provided in the link. one at the top of the page and server artifacts 12746

Hey TRK! You got it working? I fixed the script fully so it is functioning on 12882.

**Comment by DevNettrix** [comment]
> Then idk why it isn't working for you, i've tested it on a fresh local server and for me its working on artifact 12882
> 
> a friend of mine is on artifact 12651 and it's working there just fine with the same file as in my pull request, i've created a release on the fix branch, maybe try downloading it through there?
> 
> https://github.com/Lap76/IIS-EmergencyLS-ELS-FiveM/releases/tag/1.0.0

Hi! If you have discord I can send you the one fully working so you can upload that?

**Comment by YordiGhurt** [comment]
i have the same Error: 

> 20:50:03 [    c-scripting-core] Creating script environments for els
> 20:50:03 [          script:els] Parsed VCF for: Example.xml
> 20:50:03 [    c-scripting-core] script error in native 0000000076a9ee1f: vector::_M_default_append
> 20:50:03 [          script:els] SCRIPT ERROR: native 0000000076a9ee1f: vector::_M_default_append
> 20:50:03 [          script:els] > handler (@els/server/init.lua:123)
> 20:50:03 [           resources] Started resource els


i have your Version uploadet to the Server

**Comment by Lap76** [comment]
It seems to be that the changes i made to the init.lua only worked on Windows my latest release also fixes it for Linux on b25770/Lin

https://github.com/Lap76/IIS-EmergencyLS-ELS-FiveM/releases/tag/1.1.0
